app/internal/world_clock.py

```diff
@@ -0,0 +1,510 @@
+    Returns:
+        str: The suitable country name.
+    """
+    data = load_city_country_data_set()
+    details = data.get(city_name.title())
+    if not details:
+        return
+    return details.get(COUNTRY_NAME_KEY_IN_CITIES)
+
+
+def get_subcountry(city_name: str) -> Optional[str]:
+    """Get the subcountry of a city.
+    Args:
+        city_name (str): The given city name.
+    Returns:
+        str: The suitable subcountry name.
+    """
+    data = load_city_country_data_set()
+    details = data.get(city_name.title())
+    if not details:
+        return
+    return details.get(SUBCOUNTRY_NAME_KEY_IN_CITIES)
+
+
+async def get_api_data(url: str) -> Optional[List[Any]]:
+    """Get the data from an API url.
+    Args:
+        url (str):  The API url.
+    Returns:
+        list: The data.
+    """
+    try:
+        async with httpx.AsyncClient() as client:
+            resp = await client.get(url)
+            return resp.json()
+    except (json.JSONDecodeError, httpx.HTTPError):
+        logging.exception("Http Error")
+
+
+async def parse_timezones_list() -> List[Tuple[str, ...]]:
+    """Parse the timezones list into pairs of continent-place.
+    Returns:
+        list: The parsed data.
+    """
+    timezones = await get_api_data(TIMEZONES_BASE_URL)
+    return [tuple(timezone.split(PATH_SEPARETOR)) for timezone in timezones]
+
+
+async def get_timezones_parts(part: str) -> List[str]:
+    """Get a given part of the timezones.
+    Args:
+        part (str):  The given part.
+    Returns:
+        list: a list of the relevant parts of the timezones list.
+    """
+    parse_list = await parse_timezones_list()
+    return [timezone[TIMEZONE_PARTS_MAP[part]] for timezone in parse_list]
+
+
+def standardize_country_or_place(place_name: str) -> Optional[str]:
+    """Standardize a given country or place name to the equivalent
+       name that appears in the timezone list.
+    Args:
+        place_name (str): The given country or place name.
+    Returns:
+        str: The standardized name.
+    """
+    if place_name:
+        return place_name.replace(BLANK, UNDERSCORE).title()
+
+
+async def standardize_continent(continent_name: str) -> Optional[str]:
+    """Standardize a given continent name to the equivalent name that
+       appears in the timezone list.
+    Args:
+        continent_name (str): The given continent name.
+    Returns:
+        str: The standardized name.
+    """
+    if continent_name in await get_timezones_parts(CONTINENT):
+        return continent_name
+
+
+async def search_timezone_by_just_place(place_name: str) -> Optional[str]:
+    """Search for a timezone in the timezones list by a given place name.
+    Args:
+        place_name (str): The given place name.
+    Returns:
+        str: The suitable timezone.
+    """
+    res = [
+        timezone
+        for timezone in await get_api_data(TIMEZONES_BASE_URL)
+        if place_name in timezone
+    ]
```

Comment:
This will cost an insane amount of time for no good reason :))
Move the `get_api_data` to a different line

Reply:
done